### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by shosuke-13**
### **PR Type**
enhancement, bug fix, documentation, dependencies

___

### **Description**
- Simplified training scripts by removing unnecessary loops and adjusting logging steps.
- Added new scripts for gene expression tasks with multiple models and parameters.
- Integrated S3 for saving predictions and metrics, and fixed label handling in the training script.
- Updated CI workflow to include the `dev-exp` branch.
- Corrected task type in configuration files and added `boto3` to dependencies.
- Updated README with new experiment descriptions.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>agront.sh</strong><dd><code>Simplify training script and adjust logging steps</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/agront.sh

<li>Removed loop for different seeds in training script.<br> <li> Increased <code>--logging_steps</code> from 1000 to 10000.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-5edeb7d7dd3cb0969908fd98a944b128c37017441f060fb04c48b5afafdca475">+46/-49</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dnabert.sh</strong><dd><code>Adjust logging steps in DNABERT script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/dnabert.sh

- Increased `--logging_steps` from 1000 to 10000.

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-4ec35a4778557b959afa3a5aecf8ffd8c2ac433eacad7080bd9a1aafc80b954b">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_gene.sh</strong><dd><code>Add script for gene expression task with multiple models</code>&nbsp; </dd></summary>
<hr>

finetune/scripts/dok/exp2_gene.sh

<li>Added new script for gene expression task.<br> <li> Configured training with multiple models and parameters.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-c2529b83453a1b59ef4710c7c8a2bcac5484881b394e277ce774d1d210d01a2e">+38/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_lncrna.sh</strong><dd><code>Update lncrna script with multiple seeds and fix model list</code></dd></summary>
<hr>

finetune/scripts/dok/exp2_lncrna.sh

<li>Removed duplicate model entry.<br> <li> Added loop for multiple seeds in training.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-c240a01021bd80f0495797ada58292c790a9cafc20bd39d705032f43b960230c">+24/-22</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_splicing.sh</strong><dd><code>Clean up model list in splicing script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/exp2_splicing.sh

- Removed duplicate model entry.

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-d96b1e011c3aa1986b78bd19f90d62f0f30f1f9df30dee3a0a4e3ab28332c873">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_tasks.sh</strong><dd><code>Clean up model list in tasks script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/exp2_tasks.sh

- Removed duplicate model entry.

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-91a001f3c05a5d11ce7eaa70d58a4b808784a40e4001a69ead955d9e8efc0cad">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>train.py</strong><dd><code>Integrate S3 for results and fix label handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/train.py

<li>Added <code>boto3</code> for S3 integration.<br> <li> Fixed label key handling in <code>tokenize_function</code>.<br> <li> Updated <code>save_results</code> to upload predictions and metrics to S3.<br> <li> Fixed <code>num_labels</code> handling for multi-variable regression.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016">+51/-9</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>docker-image.yml</strong><dd><code>Update CI workflow to include dev-exp branch</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/docker-image.yml

- Added `dev-exp` branch to CI workflow triggers.

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-dbade1e5797a7633e024a7685a43a7cbd96ae6fefc6314294807ae34cd83712c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Update README with experiment descriptions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.md

<li>Added sections for experiments on kmer tokenization and model <br>evaluation.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>pgb_tasks.yaml</strong><dd><code>Fix task type in configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/pgb_tasks.yaml

<li>Corrected task type from <code>multiple_variable_regression</code> to <br><code>multi_variable_regression</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-eabc82996fdb1fc25f12d5d1110ac5d5e61a3d0dddd7475f423be1bfc7dc70c9">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Add boto3 to requirements</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

requirements.txt

- Added `boto3` dependency.

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by shosuke-13** [comment]
Preparing PR description...

**Comment by shosuke-13** [comment]
Preparing review...

**Comment by shosuke-13** [comment]
## PR Reviewer Guide 🔍

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 4 🔵🔵🔵🔵⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Key issues to review</strong><br><br>

<a href='https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-5edeb7d7dd3cb0969908fd98a944b128c37017441f060fb04c48b5afafdca475R5-R51'><strong>Code Duplication</strong></a><br>The code for training with LoRA and IA3 is duplicated. Consider refactoring to avoid redundancy.

<a href='https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-c2529b83453a1b59ef4710c7c8a2bcac5484881b394e277ce774d1d210d01a2eR18-R18'><strong>Undefined Variable</strong></a><br>The variable `MODEL` is used but not defined. It should be replaced with `model` in line 18.

<a href='https://github.com/shosuke-13/kmer-tokenization-strategy/pull/5/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016R421-R422'><strong>Potential Bug</strong></a><br>The logic for determining `num_labels` in line 421 may lead to errors if `task_details` does not have the expected structure.

</td></tr>
</table>

**Comment by shosuke-13** [comment]
## Generating PR code suggestions


Work in progress ...<br>
